src/validator/MatchValidator.cpp

```diff
@@ -23,47 +23,113 @@ AstContext* MatchValidator::getAstContext() {
     return matchCtx_.get();
 }
 
-
 Status MatchValidator::validateImpl() {
-    auto *sentence = static_cast<MatchSentence*>(sentence_);
+    auto *sentence = static_cast<MatchSentence *>(sentence_);
     auto &clauses = sentence->clauses();
 
-    if (clauses.size() > 1) {
-        return Status::SemanticError("Multi clause MATCH not supported");
-    }
+    std::unordered_map<std::string, AliasType> *aliasesUsed = nullptr;
+    YieldColumns *prevYieldColumns = nullptr;
+    auto retClauseCtx = getContext<ReturnClauseContext>();
+    for (size_t i = 0; i < clauses.size(); ++i) {
+        switch (clauses[i]->kind()) {
```

Comment:
Why not extract some common functions for the huge switch check?

Reply:
For common codes, I have extracted them into functions. Such as `combineAliases`.  For the other codes, I coudn't extract them.